app/components/widgets/ProgressSteps.js

```diff
@@ -9,24 +11,31 @@ import iconCrossSVG from '../../assets/images/widget/cross.inline.svg';
 import iconCrossGreenSVG from '../../assets/images/widget/cross-green.inline.svg';
 import styles from './ProgressSteps.scss';
 
+// TODO: move to type folder?
+export const StepState = Object.freeze({
+  LOAD: 0,
+  PROCESS: 1,
+  ERROR: 9,
+});
+export type StepStateEnum = $Values<typeof StepState>;
+
 type Props = {
   stepsList: Array<string>,
-  progressInfo: {
-    currentStep : number, // example, 0 = pointing to stepsList[0]
-    stepState: number, // has three states, 0 = LOAD | 1 = PROCESS | 9 = ERROR
-  },
+  currentStep : number, // example, 0 = pointing to stepsList[0]
+  stepState: StepStateEnum,
```

Comment:
I don't get the change of variables in this file. Is there a reason for removing progressInfo? I think this logic of using progressInfo was replicated from other parts of the code related to HW connection, and it's used in several parts. It doesn't seem to conflict with this file, but maybe the idea was to keep the same notation.

Reply:
Because the type of `currentStep` depends on where the progress bar is used (you can imagine different processes requiring a different number of steps, all represented by their own enum type). Suppose you have the two types:
```
type Foo: {
    enum: number;
}

type MyEnum = 1 | 2 | 3;
type Bar: {
    enum: MyEnum;
}
```

You can't cast `Bar` to `Foo`. You can, however, do `Foo.enum = Bar.enum`

For this reason, we can't keep the progressInfo as previously defined without ugly code so it's easier to just flatten it.